src/orders/OrderProcessor.sol

```diff
@@ -794,7 +794,9 @@ contract OrderProcessor is
             IDShare(order.assetToken).mint(requester, unfilledAmount);
         } else {
             // Return unfilled
-            IERC20(order.paymentToken).safeTransferFrom(msg.sender, requester, unfilledAmount);
+            if (unfilledAmount > 0) {
```

Comment:
Remove unnecessary check. Unfilled must be nonzero or else order would not be cancellable.